### DISCUSSION TITLE
My Environment variables (env) are only visible in my Nuxt 3 build server.

### DISCUSSION THREAD

**Discussion by felixfavour** [nuxt/nuxt#21608] [Questions] [answered]
Hi team,

I have a long-standing issue with understanding the way environment variables work in Nuxt 3—primarily why they show up in my Vite build server, but are not accessible in my browser's log console.

I would love an explanation as to why this happens if anyone can share. I would also love to know how I can access these env variables from the browser and log them in my console.

I don't think this question needs any code attachment or something of that sort but I am happy to make up an example if need be. Thanks.

**Comment by warflash** [comment]
https://nuxt.com/docs/api/composables/use-runtime-config#access-runtime-config if you put env variables in the public object of the runtime config then you'll have access to it on client as well. e.g. `NUXT_PUBLIC_BASE_URL` will be available at `useRuntimeConfig().public.baseUrl`

**Reply by felixfavour** [reply]
Thanks for your response. I have used the public runtime config in the past, so I'm aware of how it helps ensure that variables are accesible on the browser.

However is the same possible for env variables?

**Comment by AloisSeckar** [comment, marked as answer]
`process.env` is only available for applications running in Node environment. It is not exposed to browser. If you need such values, you need to do it yourself. 

Here I believe is an article how to do it in a "raw" way using Vite or Webpack - [https://dmitripavlutin.com/environment-variables-javascript/](https://dmitripavlutin.com/environment-variables-javascript/). But I would stick with useRuntimeConfig in Nuxt.

**Reply by felixfavour** [reply]
Thanks, `process.env` only works on the node env. Got it!
This is helpful, I eventually got it to work on nuxt server directory. Used the /server/api as a "proxy".

I guess I'll stick to `runtimeConfig` for storing keys on the client-side.

**Reply by spraddles** [reply]
@felixfavour how did you manage to do this? I still can't get the `ENV` vars to work in my `/server/*` directories...

**Reply by felixfavour** [reply]
@spraddles Did you add them to your `runtimeConfig` yet via the nuxt-config file?
I added mine in this way:
```
  runtimeConfig: {
    public: {
      baseURL: process.env.BASE_URL
    },
  },
```

and used it like this in my server route:
```
export default defineEventHandler(async (event) => {
  const config = useRuntimeConfig(event)
  const authUser = await retrieveAuth(event.$fetch, `${config.public.baseURL}/user`, "STRING_PAYLOAD")
    return sendRedirect(event, '/', 302)
  }
  return sendRedirect(event, '/login', 302)
})
```

**Reply by spraddles** [reply]
I wanted to try & avoid adding `runtimeConfig` etc... and eventually I found that adding this package worked: `@nuxtjs/dotenv` - and I didn't need to register the module for build etc. it just works out of the box
